### ISSUE TITLE
C# Ternary condition operator breaks color highlighting when first expression has function right away

### ISSUE THREAD

**Issue by Apostolique** [dotnet/csharp-tmLanguage#218]
## Details

What editor are you seeing the problem in?
Visual Studio Code

What version of the editor are you using?
1.59.1

What color theme are you using?
Dark+
AposTheme

## Repro

```csharp
namespace BrokenSyntax {
    public static class Program {
        static void Main() {
            Test1 = null == null ? Math.Max(0, 1) : 2; // Broken

            Test2 = null == null ? (Math.Max(0, 1)) : 2; // Works
            Test3 = null == null ? /**/ Math.Max(0, 1) : 2; // Works
            var Test4 = null == null ? Math.Max(0, 1) : 2; // Works
        }

        public static int Test1;
        public static int Test2;
        public static int Test3;
    }
}
```

Dark+:
![image](https://user-images.githubusercontent.com/2766423/130202623-25bbea5e-7034-436f-854e-88f1afeaefc4.png)

AposTheme:
![image](https://user-images.githubusercontent.com/2766423/130202593-2a2e73bf-9d47-4a69-8c92-9a92e738e620.png)

**Comment by am11** [comment]
GitHub Linguist (repository that drives syntax highlighting in GitHub comments and source code) and VSCode both are using C# grammar from this repo.

In github/linguist repo, we have added this repo [as a git submodule](https://github.com/github/linguist/blob/0fca6f46491731d9adedc678d8663aa74d40a5a6/.gitmodules#L359-L361), so we get pure experience with all the fixes from single source of truth; this repository.

However, VSCode has manual copy of grammar file from this repo, then there are some one-off modifications on top, which makes it hard to track the changes. If something is different in GitHub and VSCode, in most cases the approach of "consuming" this repo is to blame. IMO, we should just upstream fixes here, without patching it up in the consumer repo.

**Comment by wise0704** [comment]
Fixed in #271
![image](https://github.com/dotnet/csharp-tmLanguage/assets/13545633/e34a9960-f621-425e-90f2-38342cc3e571)
